schema/policy_test.go

```diff
@@ -928,6 +984,96 @@ var _ = Describe("Policies", func() {
 				}
 				Expect(filter).To(Equal(expected))
 			})
+
+			It("Should create a correct filter for is_domain_owner", func() {
+				tenant := Tenant{ID: "test", Name: "test"}
+				testAuth = NewAuthorizationBuilder().
+					WithTenant(tenant).
+					WithRoleIDs("admin").
+					BuildScopedToTenant()
+
+				schema := getSchema("test")
+				testPolicy["condition"] = []interface{}{
+					map[string]interface{}{
+						"or": []interface{}{
+							"is_domain_owner",
+							map[string]interface{}{
+								"match": map[string]interface{}{
+									"property": "status",
+									"type":     "eq",
+									"value":    "ACTIVE",
+								},
+							},
+						},
+					},
+				}
+
+				var err error
+				policy, err = NewPolicy(testPolicy)
+				Expect(err).ToNot(HaveOccurred())
+				filter := map[string]interface{}{}
+				currCond := policy.GetCurrentResourceCondition()
+				currCond.AddCustomFilters(schema, filter, testAuth)
+				expected := map[string]interface{}{
+					"__or__": []map[string]interface{}{
+						{
+							"property": "domain_id",
+							"type":     "eq",
+							"value":    testAuth.DomainID(),
+						},
+						{
+							"property": "status",
+							"type":     "eq",
+							"value":    "ACTIVE",
+						},
+					},
+				}
+				Expect(filter).To(Equal(expected))
+			})
+
+			It("Should omit creating a filter for is_domain_owner when domain_id is missing from schema", func() {
+				tenant := Tenant{ID: "test", Name: "test"}
+				testAuth = NewAuthorizationBuilder().
+					WithTenant(tenant).
+					WithRoleIDs("admin").
+					BuildScopedToTenant()
+
+				schema := getSchema("network")
+				testPolicy["condition"] = []interface{}{
```

Comment:
This is the same condition as in the test above. Can we extract and name it?

Reply:
Done